Source/WebCore/html/ImageData.h

```diff
@@ -43,9 +43,11 @@ class ImageData : public RefCounted<ImageData> {
 public:
     WEBCORE_EXPORT static Ref<ImageData> create(Ref<ByteArrayPixelBuffer>&&);
     WEBCORE_EXPORT static RefPtr<ImageData> create(RefPtr<ByteArrayPixelBuffer>&&);
-    WEBCORE_EXPORT static RefPtr<ImageData> create(const IntSize&, PredefinedColorSpace);
+    WEBCORE_EXPORT static RefPtr<ImageData> createZeroFilled(const IntSize&, PredefinedColorSpace, ImageDataStorageFormat = ImageDataStorageFormat::Uint8);
     WEBCORE_EXPORT static RefPtr<ImageData> create(const IntSize&, ImageDataArray&&, PredefinedColorSpace);
-    WEBCORE_EXPORT static ExceptionOr<Ref<ImageData>> createUninitialized(unsigned rows, unsigned pixelsPerRow, PredefinedColorSpace defaultColorSpace, std::optional<ImageDataSettings> = std::nullopt);
+
+    WEBCORE_EXPORT static ExceptionOr<Ref<ImageData>> createZeroFilled(unsigned sw, unsigned sh, PredefinedColorSpace defaultColorSpace, std::optional<ImageDataSettings> = std::nullopt);
```

Comment:
I've tried that, but:
- Some generated JS binding code provides an `optional< ImageDataSettings>`, so we must still accept that.
- `ImageDataSettings::colorSpace` is itself `optional`, so every caller would have to ensure it's provided if needed.
- The behavior of defaultColorSpace+optional settings is given in the specs, so I think it's good to have it centralized in one place, rather than trying to replicate it at multiple call sites.
- Because of the above, there were a few places that needed some fiddly code, something like `{ { colorSpace } }`, or worse:
```
        if (!settings)
            settings = { { m_settings.colorSpace } };
        else if (!settings->colorSpace)
            settings->colorSpace = m_settings.colorSpace;
```

So I've gone with:
- One `Create` with both,
- and one `Create` with just `optional<ImageDataSettings>`, I think it's clearer at all call sites (and bonus: the implementation is trivial).